scripts/generate-method-action-types.ts

```diff
@@ -0,0 +1,536 @@
+  const { check, fix } = await yargs(process.argv.slice(2))
+    .option('check', {
+      type: 'boolean',
+      description: 'Check if generated action type files are up to date',
+      default: false,
+    })
+    .option('fix', {
+      type: 'boolean',
+      description: 'Generate/update action type files',
+      default: false,
+    })
+    .help().argv;
+
+  return { check, fix };
+}
+
+/**
+ * Checks if generated action types files are up to date.
+ *
+ * @param controllers - Array of controller information objects.
+ */
+async function checkActionTypesFiles(
+  controllers: ControllerInfo[],
+): Promise<void> {
+  let hasErrors = false;
+
+  for (const controller of controllers) {
+    const outputDir = path.dirname(controller.filePath);
+    const outputFile = path.join(
+      outputDir,
+      `${controller.name}-method-action-types.ts`,
+    );
+
+    const expectedContent = await generateActionTypesContent(
+      controller,
+      outputFile,
+    );
+
+    try {
+      const actualContent = await fs.promises.readFile(outputFile, 'utf8');
+
+      if (actualContent !== expectedContent) {
+        console.error(
+          `❌ ${controller.name}-method-action-types.ts is out of date`,
+        );
+        hasErrors = true;
+      } else {
+        console.log(
+          `✅ ${controller.name}-method-action-types.ts is up to date`,
+        );
+      }
+    } catch (error) {
+      if ((error as NodeJS.ErrnoException).code === 'ENOENT') {
+        console.error(
+          `❌ ${controller.name}-method-action-types.ts does not exist`,
+        );
+      } else {
+        console.error(
+          `❌ Error reading ${controller.name}-method-action-types.ts:`,
+          error,
+        );
+      }
+      hasErrors = true;
+    }
+  }
+
+  if (hasErrors) {
+    console.error('\n💥 Some action type files are out of date or missing.');
+    console.error(
+      'Run `yarn generate-method-action-types --fix` to update them.',
+    );
+    process.exitCode = 1;
+  } else {
+    console.log('\n🎉 All action type files are up to date!');
+  }
+}
+
+/**
+ * Main entry point for the script.
+ */
+async function main() {
+  const { fix } = await parseCommandLineArguments();
+
+  console.log('🔍 Searching for controllers with MESSENGER_EXPOSED_METHODS...');
+
+  const controllers = await findControllersWithExposedMethods();
+
+  if (controllers.length === 0) {
+    console.log('⚠️  No controllers found with MESSENGER_EXPOSED_METHODS');
+    return;
+  }
+
+  console.log(
+    `📦 Found ${controllers.length} controller(s) with exposed methods`,
+  );
+
+  if (fix) {
+    for (const controller of controllers) {
+      console.log(`\n🔧 Processing ${controller.name}...`);
+      await generateActionTypesFile(controller);
+      console.log(`✅ Generated action types for ${controller.name}`);
+    }
+
+    console.log('\n🎉 All action types generated successfully!');
+  } else {
+    // Default mode, --check mode, or no mode: check files
```

Comment:
If we do not have a default mode as per my suggestion above, then maybe we can take away that part of the comment here?

``` suggestion
    // --check mode: check files